ch-remote: fix tls-dir handling

phip1611 opened the pull request:
Follow-up of d5e345b20882878440f082e43d33296283fd8847.

scholzp commented:
Tremendous work! :rocket: 

But can you please give some context? Why did the underscore break things?

Edit: To clarify, by this I mean that it would be nice to have some answer to `Why?` in the commit message. Thanks!

amphi commented on src/bin/ch-remote.rs:
```diff
@@ -533,9 +533,9 @@ fn rest_api_do_command(matches: &ArgMatches, socket: &mut UnixStream) -> ApiResu
+                    .subcommand_matches("receive-migration")
                     .unwrap()
-                    .get_one::<PathBuf>("tls_dir")
+                    .get_one::<PathBuf>("tls-dir")
```
This change would also require changes in libvirt, right? And given that there is no consistent usage of snake case or kebab case, do we really want that? Just look a few lines higher and you find `receive_migration_config`.

phip1611 replied:
This doesn't affect libvirt at all. This is ch-remote, not the Cloud Hypervisor HTTP API.

As ch-remote unfortunately uses clap's builder API rather than the derive API (shame 😞), some things are a little unclear at first glance, for example what is exposed to the user. Here, `tls-dir` refers to `--tls-dir <path>` in the CLI. You can verify that by running `ch-remote receive-migration --help`.

I switched to snake-case so that we can specify `--tls-dir` and not `--tls_dir` which is very untypical in CLIs, especially in Rust. 

What you are referring to with `receive_migration_config` is not actually an exposed `--option <val>` but refers to a positional argument. Therefore this is not so confusing, although I don't like it either:

<img width="839" height="228" alt="image" src="https://github.com/user-attachments/assets/0ea98bc8-8500-4235-aee7-8e98b1bc57b9" />



Again, if ch-remote would use clap's derive API, things would be much simpler and we would not have to deal with those annoying IDs at all and snake-case would be the default.

amphi replied:
Just for clarification: You could also write

```rust
                Arg::new("tls_dir")
                    .long("tls-dir")
                    .help("directory with TLS certificates")
                    .num_args(1),
```

right? Or would this break anything?

But just keep in mind that this requires changes to libvirt

phip1611 replied:
> But just keep in mind that this requires changes to libvirt

no, it doesn't. This is ch-remote, not the Cloud Hypervisor API.

phip1611 replied:
This is the difference. There is no clear policy in CHV for this, I think. I think it is just technical yolo land - it would go away (or at least be consistent) with the derive API. I can also switch that to `_` - no strong opinion. As long as the value in `.long()` uses a dash.

<img width="1281" height="315" alt="image" src="https://github.com/user-attachments/assets/f03201b8-d6f8-44ac-a6ab-c317c57b408e" />

phip1611 committed:
ch-remote: remove missing `--tls-dir` option to migration commands

This was missing. We tried to access the option from the parsed args but
it is not yet known to clap that clap should parse these options.

Follow-up of d5e345b20882878440f082e43d33296283fd8847.

Signed-off-by: Philipp Schuster <philipp.schuster@cyberus-technology.de>
On-behalf-of: SAP philipp.schuster@sap.com

phip1611 committed:
ch-remote: tls_dir -> tls-dir

Rename to the same name that the argument parser uses. This is also
consistent with the rest of the CLI, where underscores are unusual.

Signed-off-by: Philipp Schuster <philipp.schuster@cyberus-technology.de>
On-behalf-of: SAP philipp.schuster@sap.com

phip1611 commented:
> Edit: To clarify, by this I mean that it would be nice to have some answer to `Why?` in the commit message. Thanks!

hahah, you've got me 😀 fixed
